File: CleverFerretV2/feature/ai/src/main/java/com/cleverferret/v2/feature/ai/budgeting/BudgetDecision.java
package com.cleverferret.v2.feature.ai.budgeting;

public record BudgetDecision(boolean approved, String reason) {
    public static BudgetDecision approved() {
        return new BudgetDecision(true, "approved");
    }

    public static BudgetDecision rejected(String reason) {
        return new BudgetDecision(false, reason);
    }
}

File: CleverFerretV2/feature/ai/src/main/java/com/cleverferret/v2/feature/ai/budgeting/BudgetDecision.kt
package com.cleverferret.v2.feature.ai.budgeting

data class BudgetDecision(val approved: Boolean, val reason: String) {
    companion object {
        fun approved() = BudgetDecision(true, "approved")
        fun rejected(reason: String) = BudgetDecision(false, reason)
    }
}
